data/movements/scripts/quests/dreamer challenge/riddle.lua

```diff
@@ -1,37 +1,17 @@
 function onStepIn(creature, item, position, fromPosition)
 	item:transform(425)
 
-	if item.actionid == 2245 then
-		local new_position = Position(0, position.y, position.z)
-		for i = 5, 0, -1 do
-			new_position.x = position.x + 2 + i
-			local tile = new_position:getTile()
-			if i == 5 then
-				new_position.x = position.x + 2
-			else
-				new_position.x = new_position.x + 1
-			end
+	local index = item.actionid == 2245 and 'x' or 'y'
 
-			local itemCount = tile:getDownItemCount()
-			if itemCount > 0 then
-				tile:getThing(tile:getTopItemCount() + tile:getCreatureCount() + itemCount):moveTo(new_position)
-			end
-		end
-	elseif item.actionid == 2246 then
-		local new_position = Position(position.x, 0, position.z)
-		for i = 5, 0, -1 do
-			new_position.y = position.y + 2 + i
-			local tile = new_position:getTile()
-			if i == 5 then
-				new_position.y = position.y + 2
-			else
-				new_position.y = new_position.y + 1
-			end
-
-			local itemCount = tile:getDownItemCount()
-			if itemCount > 0 then
-				tile:getThing(tile:getTopItemCount() + tile:getCreatureCount() + itemCount):moveTo(new_position)
-			end
+	local new_position = Position(position.x, position.y, position.z)
+	
+	for i = 1, 6 do
+		new_position[index] = position[index] + 2 + i - 1
```

Comment:
yeah i just like to keep the numbers so they can keep their meaning... when reading that script i realize immediately what + 2 stands for (across the walls) but seeing just + 1 would confuse me and i wouldn't be able to connect it without thinking too hard....

it's up to you though, should i make your suggested change?